### ISSUE TITLE
Tor support for desktop app

### ISSUE THREAD

**Issue by chirhonul** [tutao/tutanota#1135]
**Describe the bug**

This is not a bug report but a feature request. Thanks for the work in creating desktop clients (https://tutanota.com/blog/posts/desktop-clients), but as far as I could tell there's no simple way to make the client use a SOCKS5 proxy to connect to the Internet, e.g to the default address of `127.0.0.1:9150` when using Tor. Ideally the desktop client would notice that there is such a proxy available and prompt the user whether to use it by default. 

**To Reproduce**

1. Start `./tutanota-desktop-linux.AppImage` version `3.47.4` when routing all traffic via Tor
2. Attempt to log in
3. Receive `Could not reach server, looks like you are offline. Please try again later.`

**Expected behavior**

Client can reach servers and log in, either with zero configuration for the user running SOCKS5 proxy on `127.0.0.1:9150`, or after specifying a command line argument to explicitly state that SOCKS5 should be used and the address to use to reach the proxy.

**Desktop (please complete the following information):**
 - OS: Linux
 - Desktop Environment, if applicable (on Linux): GNOME

**Comment by mvfauafk** [comment]
you can start with that:

/opt/tutanota-desktop-linux/tutanota-desktop-linux.AppImage --proxy-server="socks://localhost:9050"

but it would be easier with a function implemented on the application.

**Comment by hiiamboris** [comment]
I have found [this guide](https://www.chromium.org/developers/design-documents/network-stack/socks-proxy) useful.
TL;DR the option `--proxy-server="socks5://127.0.0.1:port"` has worked for me for Privoxy+Tor solution on Windows. While `localhost:port` won't work without the extra `--host-resolver-rules` option.

**Comment by sang-shelton** [comment]
I am doing
`./tutanota-desktop-linux.AppImage --host-resolver-rules --proxy-server="socks://localhost:9050"`
and still cannot login, tor is running fine on my machine, yet i am getting this error:

<details><summary>Error</summary>
<p>

```
[ProtocolProxy] error for GET https://app.tuta.com/rest/sys/session/NFk5FD3--Z-9/VcyywNxjgbHmYNNLhqE8ycWcR7OAzPydzyfCL_QtvoY:
[ProtocolProxy] TypeError: fetch failed
    at /tmp/.mount_tutano1w5VIt/resources/app.asar/desktop/DesktopMain.js:10:29050
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /tmp/.mount_tutano1w5VIt/resources/app.asar/desktop/DesktopMain.js:10:29558
    at async AsyncFunction.<anonymous> (node:electron/js2c/browser_init:2:47442) {
  [cause]: AggregateError
      at internalConnectMultiple (node:net:1114:18)
      at internalConnectMultiple (node:net:1177:5)
      at Timeout.internalConnectMultipleTimeout (node:net:1687:3)
      at listOnTimeout (node:internal/timers:575:11)
      at process.processTimers (node:internal/timers:514:7) {
    code: 'ETIMEDOUT',
    [errors]: [ [Error], [Error] ]
  }
}
[ProtocolProxy] {"name":"TypeError","message":"fetch failed","stack":"TypeError: fetch failed\n    at /tmp/.mount_tutano1w5VIt/resources/app.asar/desktop/DesktopMain.js:10:29050\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async /tmp/.mount_tutano1w5VIt/resources/app.asar/desktop/DesktopMain.js:10:29558\n    at async AsyncFunction.<anonymous> (node:electron/js2c/browser_init:2:47442)"}
[ProtocolProxy] failed after 266ms
```

</p>
</details>
